src/schema/slack/functions/_scripts/src/templates/template_function.ts

```diff
@@ -0,0 +1,153 @@
+import { isCustomType } from "../../../../../../types/mod.ts";
+import SchemaTypes from "../../../../../schema_types.ts";
+import SlackSchemaTypes from "../../../../schema_types.ts";
+import { InternalSlackTypes } from "../../../../types/custom/mod.ts";
+import {
+  FunctionParameter,
+  FunctionProperties,
+  FunctionProperty,
+  FunctionRecord,
+} from "../types.ts";
+import {
+  autogeneratedComment,
+  getSlackCallbackId,
+  renderTypeImports,
+} from "./template_utils.ts";
+import { AllowedTypeValue, AllowedTypeValueObject } from "./types.ts";
+
+const typeMap: Record<string, AllowedTypeValueObject> = {
+  SchemaTypes,
+  SlackTypes: SlackSchemaTypes,
+  InternalSlackTypes,
+};
+
+const schemaTypeMap = Object.entries(typeMap).reduce<Record<string, string>>(
+  (acc, [schemaKey, schemaTypes]) => {
+    for (const schemaType in schemaTypes) {
+      const value: AllowedTypeValue = schemaTypes[schemaType];
+      const type: string = isCustomType(value) ? value.id : value;
+      acc[type] = `${schemaKey}.${schemaType}`;
+    }
+    return acc;
+  },
+  {},
+);
+
+const propertyToTypeScript = (
+  property: FunctionProperty,
+): string => {
+  const typescript = [];
+  typescript.push(
+    `type: ${schemaTypeMap[property.type]}`,
+  );
+  if (property.description) {
+    typescript.push(`description: "${property.description}"`);
+  }
+  if (property.title) {
+    typescript.push(`title: "${property.title}"`);
+  }
+  if ("items" in property) {
+    typescript.push(`items: ${propertyToTypeScript(property.items)}`);
+  }
```

Comment:
Array items can't recurse directly, they need to be wrapped in an object

```suggestion
  if ("items" in property) {
    typescript.push(`items: {${propertyToTypeScript(property.items)}}`);
  }
```

Reply:
Good catch 💯  I made this more generic to `properties`

Reply:
I resolved this by wrapping the property object with `{  }` on L63